doc/filters/merge.rst

```diff
@@ -40,6 +40,20 @@ overridden.
 
         {# items now contains { 'apple': 'fruit', 'orange': 'fruit' } #}
         
+.. tip::
+
+    If the values or keys you want to merge are variable, you need to enclose them into parenthesis :
+
+    .. code-block:: jinja
+        {% set mergedItems = {} %}
+        {% set items = { 'apple': 'fruit', 'orange': 'fruit' } %}
+
+        {% for key, value in items %} 
+            {% set mergedItems = mergedItems|merge({ (key): (value) }) %}
```

Comment:
It's already implicit in the note below about the fact that `array_merge` is used behind the scene. But we can indeed add a sentence in the note about how that works. Can you do that?